### PR TITLE
feat(AIP-148): lint IP Address field format

### PR DESCRIPTION
**Pull request by noahdietz**
As mandated in AIP-148, and IP Address field must have `(google.api.field_info).format` with one of the IP Address formats. This rule enforces that.

### CODE REVIEW

File: rules/aip0148/ip_address_format.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+var ipAddressFormats = []annotations.FieldInfo_Format{
+	annotations.FieldInfo_IPV4,
+	annotations.FieldInfo_IPV6,
+	annotations.FieldInfo_IPV4_OR_IPV6,
+}
+
+var ipAddressFormat = &lint.FieldRule{
+	Name: lint.NewRuleName(148, "ip-address-format"),
+	OnlyIf: func(fd *desc.FieldDescriptor) bool {
```
Comment (andrei-scripniciuc) [inline rules/aip0148/ip_address_format.go:34]: we should also check that the field is of type string, to comply with go/aip/148#well-known-string-fields. I'm not sure if we should throw an error for this for an already existing field, where changing the type is a breaking change. Do we have any precedents for this kind of change to the linter?

Comment (noahdietz) [reply rules/aip0148/ip_address_format.go:34]: We already have a linter rule for that guidance in AIP-202 :) https://linter.aip.dev/202/string-only-format

Comment (noahdietz) [reply rules/aip0148/ip_address_format.go:34]: Oh I misunderstood the comment. I think that is a good idea, thanks.

Comment (noahdietz) [reply rules/aip0148/ip_address_format.go:35]: Done!

**Review by andrei-scripniciuc — APPROVED** [review]

**Pull request merged**